lib/internal/streams/web.js

```diff
@@ -0,0 +1,4632 @@
+'use strict';
+
+/* eslint-disable no-use-before-define */
+
+const {
+  ArrayBuffer,
+  ArrayBufferPrototype,
+  ArrayBufferPrototypeSlice,
+  ArrayPrototypePush,
+  ArrayPrototypeShift,
+  DataViewCtor,
+  FunctionPrototypeBind,
+  FunctionPrototypeCall,
+  MathMax,
+  MathMin,
+  Number,
+  NumberIsInteger,
+  NumberIsNaN,
+  ObjectCreate,
+  ObjectDefineProperties,
+  ObjectGetOwnPropertyDescriptors,
+  ObjectGetPrototypeOf,
+  ObjectSetPrototypeOf,
+  Promise,
+  PromisePrototypeCatch,
+  PromisePrototypeThen,
+  PromiseResolve,
+  PromiseReject,
+  PromiseAll,
+  ReflectGet,
+  Symbol,
+  SymbolAsyncIterator,
+  Uint8Array,
+} = primordials;
+
+const {
+  codes: {
+    ERR_ILLEGAL_CONSTRUCTOR,
+    ERR_INVALID_ARG_VALUE,
+    ERR_INVALID_ARG_TYPE,
+    ERR_INVALID_STATE,
+    ERR_INVALID_THIS,
+  },
+} = require('internal/errors');
+
+const {
+  DOMException,
+} = internalBinding('messaging');
+
+const {
+  copyArrayBuffer,
+  detachArrayBuffer
+} = internalBinding('buffer');
+
+const {
+  isArrayBufferView,
+  isDataView,
+  isPromise,
+} = require('util/types');
+
+const {
+  createDeferredPromise,
+  emitExperimentalWarning,
+  customInspectSymbol: kInspect,
+} = require('internal/util');
+
+const {
+  inspect,
+} = require('util');
+
+const {
+  serialize,
+  deserialize,
+} = require('v8');
+
+const {
+  validateObject,
+} = require('internal/validators');
+
+const {
+  kAborted,
+} = require('internal/abort_controller');
+
+const {
+  MessageChannel,
+} = require('internal/worker/io');
+
+const {
+  kDeserialize,
+  kTransfer,
+  kTransferList,
+  JSTransferable,
+} = require('internal/worker/js_transferable');
+
+const {
+  getPromiseDetails,
+  kPending,
+} = internalBinding('util');
+
+const {
+  queueMicrotask,
+} = require('internal/process/task_queues');
+
+const assert = require('internal/assert');
+
+emitExperimentalWarning('stream/web');
+
+const kAbort = Symbol('kAbort');
+const kCancel = Symbol('kCancel');
+const kClose = Symbol('kClose');
+const kCloseSentinel = Symbol('kCloseSentinel');
+const kChunk = Symbol('kChunk');
+const kError = Symbol('kError');
+const kPull = Symbol('kPull');
+const kState = Symbol('kState');
+const kType = Symbol('kType');
+
+const AsyncIteratorPrototype =
+  ObjectGetPrototypeOf(
+    ObjectGetPrototypeOf(
+      async function* () {}).prototype);
```

Comment:
this seems useful to move to primordials

Reply:
Yeah, except that it doesn't seem to actually work the way we need it too. This is one that I'm going to be investigating further because the `AsyncIteratorPrototype` here because it's not quite correct.

Reply:
Ok, figured it out. It's still not perfect given that the `AsyncIteratorPrototype` is itself susceptible to prototype polution but at least the brand check now works.